### PR TITLE
WIFI-2502 Fix WM crash due to invalid data

### PR DESCRIPTION
**Pull request by RickSommerville**
Signed-off-by: Rick Sommerville <rick.sommerville@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 1]
```diff
@@ -1335,7 +1335,11 @@ bool target_vif_config_del(const struct schema_Wifi_VIF_Config *vconf)
...
 
 		ifname = uci_lookup_option_string( vif_ctx, s, "ifname" );
-		if (!strcmp(ifname,vconf->if_name)) {
+		if (ifname == NULL) {
+			/* Delete this section because it doesn't have an ifname - it is invalid */
+			uci_section_del(vif_ctx, "vif", "wireless", (char *)s->e.name, "wifi-iface");
+		} else if (!strcmp(ifname,vconf->if_name)) {
+			/* Delete this section because it matches the if_name we are trying to delete */
 			uci_section_del(vif_ctx, "vif", "wireless", (char *)s->e.name, "wifi-iface");
 			break;
```
Comment (shimmy568) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1344]: Not part of the PR, but this break seems redundant

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 2]
```diff
@@ -1335,7 +1335,11 @@ bool target_vif_config_del(const struct schema_Wifi_VIF_Config *vconf)
 		if (strcmp(s->type, "wifi-iface")) continue;
 
 		ifname = uci_lookup_option_string( vif_ctx, s, "ifname" );
-		if (!strcmp(ifname,vconf->if_name)) {
+		if (ifname == NULL) {
```
Comment (shimmy568) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1338]: Seems like we're duplicating the `uci_section_del` call, may as well merge the `if` and `else if` statements

```suggestion
		if (ifname == NULL || !strcmp(ifname,vconf->if_name)) {
```

**Review by shimmy568 — CHANGES_REQUESTED** [review]
Just noticed a few nit-picky things that could be fixed. Nothing major

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1344]: The break actually exits the uci_foreach_safe() loop.  Once we have found what we want to delete and deleted it, stop looking.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 2 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1338]: I only want the break in the 2nd case.

**Review by shimmy568 — APPROVED** [review]
Looks good to me

**Comment by RickSommerville** [comment]
Merged 28bc1dae
